mobly/controllers/android_device_lib/jsonrpc_client_base.py

```diff
@@ -0,0 +1,273 @@
+#/usr/bin/env python3.4
+#
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Base class for clients that communicate with apps over a JSON RPC interface.
```

Comment:
Strongly disagree with a customizable format; that defeats the whole point of a common protocol that anything can speak. :) That's why we went with sl4a's protocol in the first place.